CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+cmake_minimum_required (VERSION 3.11)
+project (Lerc)
+
+SET(CMAKE_CXX_FLAGS  "-g ${OPTIMIZATION_LEVEL} -pipe -Wall -Wno-unused-function -std=c++17 -Wno-deprecated-declarations -fsized-deallocation -fPIC" CACHE INTERNAL "" FORCE)
```

Comment:
Should this line instead use cmake builtins such as set(CMAKE_CXX_STANDARD 17). These flags are all clang specific. It also requires OPTIMIZATION_LEVEL to be set instead of using CMAKE_BUILD_TYPE to control symbols and optimization levels.

Reply:
Fixed. Also removed all the spare flags there